### ISSUE TITLE
Broken CSS assets emitted / omitted with wmr build

### ISSUE THREAD

**Issue by jbt** [preactjs/wmr#326]
I'm seeing some very weird behaviour when using `wmr build` with CSS assets - in some instances the assets are being incorrectly combined, or left out entirely from `dist`

## Test case

**index.html**
```html
<!DOCTYPE html>
<html lang="en">
<body>
  <script type="module" src="/index.js"></script>
</body>
</html>
```

**index.js**
```js
import './a.css';
import './b.css';
```

**a.css and b.css are both completely empty**

What appears to matter here is that `a.css` and `b.css` have exactly the same source content (the effect is the same if they both contain some actual CSS but their file contents are identical) - see below

### Expected

I expect at least one CSS file under `dist/assets` to be created

### Actual

The generated `index.[hash].js` references `/assets/a.[hash].css` but no css file is emitted.

## Remarks

I've seen variants of this where _some_ CSS is emitted but not all of it, and the JS references a missing CSS asset. I've not been able to reproduce this reliably without creating a quite complicated dependency graph though. I believe in the project where I'm seeing these errors I may have some duplicated CSS files somewhere.

I suspect this might be happening because rollup is clever and [has helpers to deduplicate emitted file names if they have the exact same content](https://github.com/rollup/rollup/blob/01a642a86edac5259e72fdb405ae4d350478328d/src/utils/FileEmitter.ts#L360) - when I enable `DEBUG` the [merging adjacent CSS assets log in optimize-graph-plugin](https://github.com/preactjs/wmr/blob/71b80e0a89060c6d8d2c3b72ff9dcbbff9ef12e8/src/plugins/optimize-graph-plugin.js#L222) logs `assets/a.92dace5d.css, assets/a.92dace5d.css` (i.e. the "a" file twice rather than "a" then "b") - and perhaps WMR isn't expecting to see multiple of the same file in `chunk.referencedFiles`

**Comment by jbt** [comment]
On further investigation, I think the original (non-reduced) issue I was seeing is slightly different, I'll open a new issue for that too once I have a test case

**Comment by developit** [comment]
@jbt curious if you think #328 fixes this, or if we're still ending up with duplicates in referencedFiles?

**Comment by jbt** [comment]
I believe this is still an issue even with the fix in #328 - from what I can tell, the reduced case here isn't affected by that issue (I reduced my test case so far I skipped right over the bug I was trying to find and came across a different one!)

Actually yes - I just tried running with my regex fix but with a.css and b.css equal and the duplicate is still showing up in referencedFiles.

I assume the fix to this one would be to deduplicate the `toMerge` array in `mergeAdjacentCss` - but I wasn't entirely sure the safest way to do that (since with CSS it matters what order things are included so it's not as simple as only taking the first occurrence)

**Comment by jbt** [comment]
I guess the empty-files case is a bit extreme and contrived, so perhaps a more concrete example that illustrates the bug, and also the importance of preserving the CSS order:

**index.js**
```js
import './a.css';
import './b.css';
import './c.css';
```

**a.css**
```css
a { text-decoration: underline; }
```

**b.css**
```css
a { text-decoration: none; }
```

**c.css**
```css
a { text-decoration: underline; }
```
(i.e. same as a.css)

It's still a bit contrived, but perhaps the a.css and b.css imports are buried in the dependency tree and you're importing c.css last to ensure your CSS overrides definitely take precedence

Currently with this input, WMR doesn't output any CSS at all since it's accidentally deleting the asset for `a.css` - which is the one everything's being combined into.

But it'd also be important not to simply deduplicate `toMerge` by first occurrence - the contents of `c.css` need to come last to ensure the correct final value of `text-decoration: underline`. It _might_ be safe to use the _last_ occurrence of each, but I don't know if there may be more edge cases I'm missing there

**Comment by JoviDeCroock** [comment]
Fixed in #385
